src/tunnelbridge_cmd.cpp

```diff
@@ -1308,7 +1308,22 @@ static void DrawTile_TunnelBridge(TileInfo *ti)
 			const RoadTypeInfo *tram_rti = tram_rt == INVALID_ROADTYPE ? nullptr : GetRoadTypeInfo(tram_rt);
 			uint sprite_offset = DiagDirToAxis(tunnelbridge_direction) == AXIS_X ? 1 : 0;
 
-			DrawRoadOverlays(ti, PAL_NONE, road_rti, tram_rti, sprite_offset, sprite_offset);
+			/* Road underlay takes precedence over tram */
+			if (road_rti != nullptr) {
+				if (road_rti->UsesOverlay()) {
+					SpriteID ground = GetCustomRoadSprite(road_rti, ti->tile, ROTSG_TUNNEL);
+					if (ground != 0) DrawGroundSprite(ground + tunnelbridge_direction, PAL_NONE);
+				}
+			} else {
```

Comment:
What happens if the road type use overlays but doesn't provide tunnel sprites?

Reply:
Original tunnel ground (with entrance and road) is still drawn before tunnel underlay, so default road is visible if no tunnel specific sprites are provided.

Reply:
As I mentioned at #10208, continuing to display the standard road overlay if the tunnel overlay action 3 is not defined would retain the old behaviour which may be expected by existing newgrfs. OTOH, displaying no overlay if the action 3 is not used would better mirror the rail tunnel spec. Which is better?

Reply:
It's easy to restore old behaviour when tunnel underlay sprites are not provided (it's controlled by a single boolean, fixed to `false` for now).
And I agree it might be better for backward compatibility with existing newgrf.